Add an `--allow-xray-errors` option

Thanks for a great tool. Earlier this week we've been getting some 504s from xray's servers which failed our tests. 

When publishing fails because the server is down or for any other reason, we prefer to get a warning and let the test pass so our war room won't get flooded. If you like the idea, I'd be happy to help with a PR.

Sure, here are the logs:

```
[10:44:32.997] [ERROR] HTTPError: Could not post to JIRA service at [MASKED]api/v2/import/execution. Response status code: 504 [pytest_xray.xray_publisher:xray_publisher.py:122]
Traceback (most recent call last):
  File "/builds/account/rider/via-rider-account/tests/black_box/.venv/lib/python3.13/site-packages/pytest_xray/xray_publisher.py", line 118, in _send_data
    response.raise_for_status()
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/builds/account/rider/via-rider-account/tests/black_box/.venv/lib/python3.13/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: [MASKED]api/v2/import/execution
{"timestamp":"2025-05-11T10:44:32.997","level":"ERROR","logger":"pytest_xray.xray_publisher","message":"HTTPError: Could not post to JIRA service at https:\/\/xray.cloud.getxray.app\/api\/v2\/import\/execution. Response status code: 504","lineno":122,"filename":"xray_publisher.py","process":210,"thread":140625320113984,"baggage":{"context_id":"545f6ff6-b3a1-4ddc-8827-235c749096b2"},"traceback":"Traceback (most recent call last):\n  File \"\/builds\/account\/rider\/via-rider-account\/tests\/black_box\/.venv\/lib\/python3.13\/site-packages\/pytest_xray\/xray_publisher.py\", line 118, in _send_data\n    response.raise_for_status()\n    ~~~~~~~~~~~~~~~~~~~~~~~~~^^\n  File \"\/builds\/account\/rider\/via-rider-account\/tests\/black_box\/.venv\/lib\/python3.13\/site-packages\/requests\/models.py\", line 1024, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\nrequests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https:\/\/xray.cloud.getxray.app\/api\/v2\/import\/execution","service.name":"context-logger","team":"N\/A"}
Traceback (most recent call last):
  File "/builds/account/rider/via-rider-account/tests/black_box/.venv/lib/python3.13/site-packages/pytest_xray/xray_publisher.py", line 118, in _send_data
    response.raise_for_status()
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/builds/account/rider/via-rider-account/tests/black_box/.venv/lib/python3.13/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
```

--- a/src/pytest_xray/xray_publisher.py
+++ b/src/pytest_xray/xray_publisher.py
@@ -1,3 +1,4 @@
+import contextlib
 import json
 import logging
 import os
@@ -120,10 +121,11 @@ class XrayPublisher:
                 err_message = (f'HTTPError: Could not post to JIRA service at {url}. '
                                f'Response status code: {response.status_code}')
                 _logger.exception(err_message)
-                if 'error' in response.json():
-                    server_return_error = f"Error message from server: {response.json()['error']}"
-                    err_message += '\n' + server_return_error
-                    _logger.error(server_return_error)
+                with contextlib.suppress(requests.exceptions.JSONDecodeError):
+                    if 'error' in response.json():
+                        server_return_error = f"Error message from server: {response.json()['error']}"
+                        err_message += '\n' + server_return_error
+                        _logger.error(server_return_error)
                 raise XrayError(err_message) from exc
             return response.json()
 

